### PR TITLE
Bluebutton-709: Made database schema changes for new bene fields

### PR DESCRIPTION
**Pull request by dho937**
This PR is strictly database schema changes to the FHIR PostgreSQL database for the additional RIF beneficiary fields from the BSF file that are being added to the Beneficiaries table.   These changes ideally can be migrated and implemented before we start processing the new RIF Beneficiary layout as this PR deals with new database fields ONLY.

### CODE REVIEW

**Comment by dho937** [comment]
This PR has just the database schema changes to the PostgreSQL FHIR database to handle the additional BSF beneficiary fields that we will be adding to the Beneficiaries table.   A new table called MedicareBeneficiaryIdHistory will also be created for another new one-time feed as part of this PR.

File: bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Creates the MedicareBeneficiaryIdHistory table required by the application. This
+ * schema is derived from the JPA metadata, via the 
+ * gov.hhs.cms.bluebutton.data.pipeline.rif.schema.HibernateSchemaPrinter 
+ * utility.
+ */
+
+${logic.tablespaces-escape} SET default_tablespace = fhirdb_ts2;
+
+create table "MedicareBeneficiaryIdHistory" (
+  "medicareBeneficiaryIdKey" numeric not null,
```
Comment (karlmdavis) [inline bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql]: Is this going to be an auto-generated sequence, or is it coming from the RIF? If it's auto-generated, you'll need a `create sequence` similar to what we have in https://github.com/CMSgov/bluebutton-data-model/blob/master/bluebutton-data-model-rif/src/main/resources/db/migration/V9__Create_beneficiary_history_table.sql

File: bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "beneficiaryId" numeric,
```
Comment (karlmdavis) [inline bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql]: No foreign key here, and I see that we don't have one on `BeneficiariesHistory.beneficiaryId`, either. Any idea why?

File: bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Creates the MedicareBeneficiaryIdHistory table required by the application. This
+ * schema is derived from the JPA metadata, via the 
+ * gov.hhs.cms.bluebutton.data.pipeline.rif.schema.HibernateSchemaPrinter 
+ * utility.
+ */
+
+${logic.tablespaces-escape} SET default_tablespace = fhirdb_ts2;
+
+create table "MedicareBeneficiaryIdHistory" (
```
Comment (karlmdavis) [inline bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql]: Also wondering if any of these should be `not null`?

File: bluebutton-data-model-rif/src/main/resources/db/migration/V10__Add_beneficiary_beneficiaryHistory_columns.sql [hunk 4]
```diff
@@ -0,0 +1,186 @@
+/*
+ * Alters the Beneficiaries and Beneficiaries History table to add additional BSF columns 
+ *
```
Comment (karlmdavis) [inline bluebutton-data-model-rif/src/main/resources/db/migration/V10__Add_beneficiary_beneficiaryHistory_columns.sql]:
```suggestion
 *
 * Note that none the modifications to existing tables here (just `Beneficiaries`, actually) will cause any locking/downtime for users of that table, per <https://www.postgresql.org/docs/9.6/sql-altertable.html>: "When a column is added with ADD COLUMN, all existing rows in the table are initialized with the column's default value (NULL if no DEFAULT clause is specified). If there is no DEFAULT clause, this is merely a metadata change and does not require any immediate update of the table's data; the added NULL values are supplied on readout, instead."
 *
```

(You'll need to reformat this one a bit, apologies.)

File: bluebutton-data-model-rif/src/main/resources/db/migration/V10__Add_beneficiary_beneficiaryHistory_columns.sql [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+ * * https://jira.cms.gov/browse/BLUEBUTTON-620
+ */
+
+alter table "Beneficiaries" add column "unhashedHicn" varchar(64);
+
+alter table "Beneficiaries" add column "medicareBeneficiaryId" varchar(11);
+alter table "Beneficiaries" add column "beneficiaryDateOfDeath"	date;
+alter table "Beneficiaries" add column "beneEnrollmentReferenceYear" numeric(4);
+alter table "Beneficiaries" add column "partAMonthsCount" numeric(3);
+alter table "Beneficiaries" add column "partBMonthsCount"	 numeric(3);
```
Comment (karlmdavis) [inline bluebutton-data-model-rif/src/main/resources/db/migration/V10__Add_beneficiary_beneficiaryHistory_columns.sql]: Could you please replace all of the tabs in here with spaces?

File: bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql (hunk 1 above)
Comment (dho937) [reply bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql]: This id key is coming from RIF.

File: bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql (hunk 2 above)
Comment (dho937) [reply bluebutton-data-model-rif/src/main/resources/db/migration/V11__Create_medicare_beneficiary_id_history_table.sql]: I created an index on the beneficiaryId later in the file.

**Pull request closed without merging**